kong.go

```diff
@@ -47,14 +47,15 @@ type Kong struct {
 	resolvers []Resolver
 	registry  *Registry
 
-	noDefaultHelp bool
-	usageOnError  bool
-	help          HelpPrinter
-	helpFormatter HelpValueFormatter
-	helpOptions   HelpOptions
-	helpFlag      *Flag
-	groups        []Group
-	vars          Vars
+	noDefaultHelp     bool
+	usageOnError      bool
```

Comment:
I think either combine these two variables into an enum (none, full, short), or ensure that both of them can't be set at once in the options, otherwise it could be confusing.

Reply:
Fair enough - I'm not too sure which one to go for:
```
kong.UsageOnError(kong.Full)
kong.UsageOnError(kong.Short)
kong.UsageOnError(kong.None)
```
is a bit of a mouth full and really it should probably be `kong.UsageOnError(kong.UsageNone)` etc., right? So even longer.

However, I don't see where the best place would be for option validation to only allow one of `kong.UsageOnError` or `kong.ShortUsageOnError`.

What do you suggest?

Reply:
We can't change the interface, I more meant that the two functions should ensure that the other isn't set, by unsetting the other's state or changing the internal values to a single enum (not exposing it).